### ISSUE TITLE
DeckKeyword constructors for Python

### ISSUE THREAD

**Issue by joakim-hove** [OPM/opm-common#1038]
The `DeckKeyword` instances which constitute the content of the `Deck` class are instantiated and assembled deep in the Parser code, when we now want to make `DeckKeyword`instances programatically from Python it is convenient to have "better" constructors. 

**DeckValue class:** Implement a small class which can contain a scalar value of type `int`, `double`,  `std::string` or "default" (no argument). 

**DeckKeyword constructor 1:** Implement a `DeckKeyword` which takes a `ParserKeyword` instance and a `std::vector<std::vector<DeckValue>>` so that the fictuous keyword 
```
TEST
    'STRING1'  1  3.14 /
    'STRING2'  *   2.71 100 /
/
```
can be instantiated as:
```
auto deck_keyword = DeckKeyword(parser_keyword, {
   {DeckValue("STRING1"), DeckValue(1), DeckValue(3.14)},
   {DeckValue("STRING2"), DeckValue(), DeckValue(2.71), DeckValue(100)}
});
```
**DecKKeyword constructor 2:** Implement a `DeckKeyword`constructor which can easily construct a data-type `DecKeyword`like `PORO` and `SATNUM` from a `std::vector<double>`or `std::vector<int>`:
```
DeckKeyword(const ParserKeyword& parser_kw, const std::vector<int>& int_data);
DeckKeyword(const ParserKeyword& parser_kw, const std::vector<double>& double_data);
```

**Comment by stefoss23** [comment]
How should the syntax look like for default values?

**Comment by joakim-hove** [comment]
> How should the syntax look like for default values?

I would say that a default constructed `DeckValue` - as the second item in the second record in the example above should be interpreted as a default - so I guess maybe the `DeckValue` class will need `DeckValue::defaulted()` method. The actual default value should come from the `ParserKeyword`; in addition I guess it should also treat the string "*" as defaulted.

**Comment by stefoss23** [comment]
OK, I will change the DeckValue class slightly. 
I think using the '*' string (DeckValue("*")?) is messy. 
I assume also the eclipse syntax for multiple defaults, example '3*', shall not be used?

**Comment by joakim-hove** [comment]
> I think using the * is messy.

I agree; but when constructing this from Python where it will typically just be:
```
deck_kw = DeckKeyword(parser_kw, [["STRING", 100, "*", 200], 
                                                            ["STRING", 1, "*", 300]])
```
we need a placeholder for the default values.

> I assume also the eclipse syntax for multiple defaults, example '3*', shall not be used?

I guess that is OK.
